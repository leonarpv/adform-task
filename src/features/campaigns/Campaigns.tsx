import React from 'react'
import { useFailedLoadingToast, useSuccessLoadingToast } from 'core/loading'
import CampaignTable from './table/CampaignTable'
import CampaignSearch from './search/CampaignSearch'
import { Campaign, CampaignFilters } from './types'
import { importCampaigns, selectCampaigns, setCampaignFilters } from './store'
import { useAppDispatch } from 'store'
import { useSelector } from 'react-redux'
import CampaignImport from './import/CampaignImport'

declare global {
    interface Window {
        AddCampaigns: (campaigns: Campaign[]) => void
    }
}

const Campaigns = () => {
    const dispatch = useAppDispatch()

    const {
        campaignsStatus,
        campaignUsersStatus,
        campaignsFiltered,
        campaignValidationError,
        filters,
    } = useSelector(selectCampaigns)

    useSuccessLoadingToast({
        status: campaignsStatus,
        message: 'Campaign Data Imported',
    })
    useFailedLoadingToast({
        status: campaignsStatus,
        message: campaignValidationError,
    })
    useFailedLoadingToast({
        status: campaignUsersStatus,
        message: 'User Fetch Failed',
    })

    window.AddCampaigns = (campaigns: Campaign[]) => {
        dispatch(importCampaigns(campaigns))
    }

    const handleChange = (filters: CampaignFilters) => {
        dispatch(setCampaignFilters(filters))
    }

    const handleImport = (campaignString: string) => {
        dispatch(importCampaigns(campaignString))
    }

    return (
        <>
            <CampaignSearch filters={filters} onChange={handleChange}>
                <CampaignImport onImport={handleImport} />
            </CampaignSearch>
            <CampaignTable
                campaigns={campaignsFiltered}
                campaignsStatus={campaignsStatus}
            />
        </>
    )
}

export default Campaigns
