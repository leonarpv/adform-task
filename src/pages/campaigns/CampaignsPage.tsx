import React from 'react'
import Campaigns from 'features/campaigns/Campaigns'

const CampaignsPage = () => {
    return <Campaigns />
}

export default CampaignsPage
