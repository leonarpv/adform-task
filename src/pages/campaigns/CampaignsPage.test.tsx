import React from 'react'
import { shallow } from 'enzyme'
import CampaignsPage from './CampaignsPage'
import Campaigns from 'features/campaigns/Campaigns'

describe('CampaignsPage', () => {
    it('should render properly', () => {
        const wrapper = shallow(<CampaignsPage />)

        expect(wrapper).toMatchSnapshot()
    })

    it('render Campaigns component', () => {
        const wrapper = shallow(<CampaignsPage />)

        expect(wrapper.find(Campaigns).exists()).toBe(true)
    })
})
