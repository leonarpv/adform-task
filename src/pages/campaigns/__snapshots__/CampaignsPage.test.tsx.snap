// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignsPage should render properly 1`] = `ShallowWrapper {}`;
